mods/default/chests.lua

```diff
@@ -274,6 +292,17 @@ function default.chest.register_chest(prefixed_name, d)
 			end
 		})
 	end
+
+	-- close opened chests on load
+	minetest.register_lbm({
+		label = "close opened chests on load",
+		name = "default:close_" .. prefixed_name:gsub(":", "_") .. "_open",
+		nodenames = {prefixed_name .. "_open"},
+		run_at_every_load = true,
+		action = function(pos, node)
+			minetest.swap_node(pos, {name = prefixed_name, param2 = node.param2})
```

Comment:
not sure if param1 actually makes a difference (engine might recalculate anyway?) but this is also a tiny but more efficient
```suggestion
			node.name = prefixed_name
			minetest.swap_node(pos, node)
```

Reply:
the current mechanism doesn't modify param1 when opening/closing the chest and i don't think it matters *what* the engine does. copy or reset to 0 would be my guess, and i think the value is probably always 0 anyway. it doesn't crash.

my personal preference is to *never* modify a passed-in table w/out copying it or documenting that it's expected to be modified. lua code can get ahold of this function through the lua API. and copying it would almost certainly erase any performance benefit.

Reply:
No need to guess here, this is documented:
 
 > If param1 or param2 is omitted, it's set to `0`.
 
 I think sfan's suggestion is cleaner though: There is no need to copy these tables since they won't be reused. This is actually pretty much engine standard: You can modify many tables (pointed things for instance) that aren't definition tables in many places since the engine takes care of the copying when calling the callbacks (which may be considered kinda dirty and inefficient). But in cases where tables are only used as a struct to pack up some values, you can modify them without issue.

Reply:
> the current mechanism doesn't modify param1 when opening/closing the chest 

```lua
				minetest.swap_node(pos, {
						name = name .. "_open",
						param2 = node.param2 })
```
Since param1 is not specified this should be the same as `param1 = 0`, which means resetting it.
Am I missing something?

> and i think the value is probably always 0 anyway

Chests have paramtype light so there will be something useful in there, not zero.

> my personal preference is to never modify a passed-in table w/out copying it

Performance is secondary here since the "improvement" probably isn't even measurable. But I wanted to note that for callbacks called by MT it's allowed to modify arguments. (Try grepping for `_copy` in builtin)

Reply:
> There is no need to copy these tables since they won't be reused

my point is that there's no way to guarantee that because things other than the engine might pass values to this code

Reply:
> > the current mechanism doesn't modify param1 when opening/closing the chest
> 
> ```lua
> 				minetest.swap_node(pos, {
> 						name = name .. "_open",
> 						param2 = node.param2 })
> ```
> 
> Since param1 is not specified this should be the same as `param1 = 0`, which means resetting it. Am I missing something?

doing some testing in-game, it seems that opening/closing a chest doesn't have any effect on the param1 value currently, even when it's non-zero, so i'd expect the code i wrote to similarly have no effect. 

> But I wanted to note that for callbacks called by MT it's allowed to modify arguments. (Try grepping for `_copy` in builtin)

interesting. but the stuff in builtin is all defensive coding, there aren't examples of modifying passed-in values that i can see. i suppose that means that it's an error to pass a table to anything and then expect to be able to re-use it afterwards? that suggests that e.g.  we should be copying `pos` before calling `swap_node` when right-clicking on a chest, and similar things? (oops accidentally rust borrow checker...)